garage/spaces/box.py

```diff
@@ -26,10 +33,21 @@ def __init__(self, low, high, shape=None):
             self.low = low + np.zeros(shape)
             self.high = high + np.zeros(shape)
 
+        if dtype is None:
+            if (self.low == 0).all() and (self.high == 255).all():
+                self.dtype = np.uint8
+                warnings.warn("garage.spaces.Box detected dtype as np.uint8.\
+                    Please provide explicit dtype.")
+            else:
+                self.dtype = np.float32
+
+        else:
+            self.dtype = dtype
+
     def sample(self):
         return np.random.uniform(
             low=self.low, high=self.high,
-            size=self.low.shape).astype(np.float32)
+            size=self.low.shape).astype(self.dtype)
```

Comment:
`high` will never be sampled if dtype is `uint8`. Please test this case and use `uniform(low, high+1)` instead.

Reply:
That's right, how about `float32`? It doesn't matter?

Reply:
i think float is not relevant because of continuity of real interval. Any number arbitrarily close to `high` can be sampled although `high` itself is never sampled, and it's almost always legit in any scenario.